packs/vsphere/actions/vm_hw_power.py

```diff
@@ -0,0 +1,25 @@
+import eventlet
+from pyVmomi import vim
+
+from vmwarelib import inventory
+from vmwarelib import checkinputs
+from vmwarelib.actions import BaseAction
+
+
+class VMPowerOn(BaseAction):
+
+    def run(self, vm_id, vm_name, power_onoff):
+        # check I have information to find a VM
+        checkinputs.vm_identifier(vm_id, vm_name)
+        # convert ids to stubs
+        vm = inventory.get_virtualmachine(self.si_content,
+                                          moid=vm_id, name=vm_name)
+        if not vm:
+            raise Exception('Error: Unable to find VM')
+        if power_onoff == "poweroff":
+            task = vm.PowerOffVM_Task()
+        else:
```

Comment:
elif power_onoff = "poweron" to make it definite and not allow arbitrary strings